### ISSUE TITLE
Validation of required values should allow empty values

### ISSUE THREAD

**Issue by easeway** [go-swagger/go-swagger#398]
## Problem statement

In an object definition, some properties (integer, string, etc.) are defined as "required".
The zero values (0, "") are all valid values of them. However, the go-swagger validation package `github.com/go-swagger/go-swagger/httpkit/validate` rejects these zero values as invalid.

It's now introducing big problems for my application when my user wants to clear some properties (set to 0, or "", they are all valid values in the application's logic) in the data models, the server simply reject the request as they are invalid.

## Swagger specification

```
swagger: '2.0'
definitions:
  Property:
    description: A property of a resource
    properties:
      propType:
        type: string
        description: discriminating type of the property
      computed:
        type: boolean
        description: is property computed
      intval:
        type: integer
        format: int64
      strval:
        type: string
    required:
      - propType
      - intval
      - strval
```

The generated code is

```
swagger:model Property
*/
type Property struct {

	/* is property computed
	 */
	Computed *bool `json:"computed,omitempty"`

	/* Intval intval

	Required: true
	*/
	Intval int64 `json:"intval,omitempty"`

	/* discriminating type of the property

	Required: true
	*/
	PropType string `json:"propType,omitempty"`

	/* Strval strval

	Required: true
	*/
	Strval string `json:"strval,omitempty"`
}
```

**Comment by easeway** [comment]
It's unreasonable to put `omitempty` for all required properties. In the json generated from the model, all these properties get lost with zero values which are valid values.

**Comment by casualjim** [comment]
duplicate of #300 ?

**Comment by easeway** [comment]
Related, but I would not say "duplicated". Because #300 is the opposite (required fields are pointers). In my situation, the fields generated are not pointers. I feel #300 should be fixed, this bug is related to validation logic, not duplicate of #300

**Comment by casualjim** [comment]
this was fixed in 42bbea36acba818fbc269e19f625316c1a0e15fc
